ext/standard/http_fopen_wrapper.c

```diff
@@ -758,25 +758,47 @@ php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 			http_header_line_length = e - http_header_line + 1;
 			http_header_line[http_header_line_length] = '\0';
 
-			if (!strncasecmp(http_header_line, "Location: ", 10)) {
+			/* The primary definition of an HTTP header in RFC 7230 states:
+			> Each header field consists of a case-insensitive field name followed
+			> by a colon (":"), optional leading whitespace, the field value, and
+			> optional trailing whitespace. */
+			char *http_header_value = http_header_line + 1;
+			while ( *http_header_value != ':' && http_header_value < http_header_line + http_header_line_length ) {
+				http_header_value++;
+			}
+			http_header_value++;
```

Comment:
Good catch. I guess it ending up as -1 could be checked below as a flag of "didn't find sane value, don't attempt to use http_header_value".